assemblies/assembly-planner-configuration.adoc

```diff
@@ -0,0 +1,1986 @@
+* Combining a time gradient dependent algorithm, most notably the Simulated Annealing algorithm, together with time spent termination. A sufficiently large difference in allocated CPU time will influence the time gradient values. Replace the Simulated Annealing algorithms with the Late Acceptance algorithm, or replace time spent termination with step count termination.
+====
+
+* `REPRODUCIBLE` mode can be slightly slower than  `NON_REPRODUCIBLE` mode.
+If your production environment can benefit from reproducibility, use this mode in production. In practice, `REPRODUCIBLE` mode uses the default, fixed random seed if no seed is specified, and it also disables certain concurrency optimizations such as work stealing.
+
+[[environmentModeProduction]]
+* `NON_REPRODUCIBLE` mode can be slightly faster than `REPRODUCIBLE` mode.
+Avoid using it during development because it makes debugging and bug fixing difficult.
+If reproducibility isn't important in your production environment, use `NON_REPRODUCIBLE` mode in production. In practice, this mode uses no fixed random seed if no seed is specified.
+
+
+[[logging]]
+=== Changing the OptaPlanner Solver logging level
+You can change the logging level in an OptaPlanner solver to review solver activity. The following list describes the different logging levels:
+
+* **error**: Logs errors, except those that are thrown to the calling code as a ``RuntimeException``.
++
+If an error occurs, OptaPlanner normally fails fast. It throws a subclass of `RuntimeException` with a detailed message to the calling code.
+It does not log it as an error to avoid duplicate log messages.
+Unles the calling code explicitly catches and eliminates that ``RuntimeException``, a ``Thread``'s default `ExceptionHandler` will log it as an error anyway.
+Meanwhile, the code is disrupted from doing further harm or obfuscating the error.
+
+* **warn**: Logs suspicious circumstances.
+* **info**: Logs every phase and the solver itself.
+* **debug**: Logs every step of every phase.
+* **trace**: Logs every move of every step of every phase.
+
+[NOTE]
+====
+Turning on `trace` logging will slow down performance considerably. However, `trace` logging is invaluable during development to discover a bottleneck.
+
+Even `debug` logging can slow down performance considerably for fast stepping algorithms such as Late Acceptance and Simulated Annealing,
+but not for slow stepping algorithms such as Tabu Search.
+
+Both trace` and `debug` logging cause congestion in multithreaded solving with most appenders.
+
+In Eclipse, `debug` logging to the console tends to cause congestion with a score calculation speeds above 10000 per second.
+Neither IntelliJ or the Maven command line suffer from this problem.
+====
+
+. Procedure
+COMMENT: How do I turn on logging?
```

Comment:
See the next chapter "Using Logback to log OptaPlanner solver activity"
We can refer it here

Reply:
This comment was outdated. I had already updated the text with clarifications from Christopher.